org.jacoco.core.test/src/org/jacoco/core/internal/analysis/filter/FilterTestBase.java

```diff
@@ -58,12 +58,30 @@ public void replaceBranches(final AbstractInsnNode source,
 		}
 	};
 
-	final void assertIgnored(Range... ranges) {
-		assertArrayEquals(ranges, ignoredRanges.toArray(new Range[0]));
+	final void assertIgnored(final MethodNode methodNode,
+			final Range... expected) {
+		assertEquals("ignored ranges",
+				rangesToString(methodNode, Arrays.asList(expected)),
+				rangesToString(methodNode, ignoredRanges));
+	}
+
+	private String rangesToString(final MethodNode m,
+			final List<Range> ranges) {
+		final StringBuilder stringBuilder = new StringBuilder();
+		for (int i = 0; i < ranges.size(); i++) {
+			final Range range = ranges.get(i);
+			stringBuilder.append("range ").append(i)
```

Comment:
Maybe add `Range.toString()`?

Reply:
@marchof it can be `Range.toString(Method)`, but not just `Range.toString` because here we need index of instruction within method - is it what you mean?

Reply:
Right... in this case I don't have a strong preference. With `Object.toString()` one could simply convert the array to a list and use its `toString()` method.